sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -162,15 +163,19 @@ public TextAnalyticsServiceVersion getServiceVersion() {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<DetectedLanguage> detectLanguage(String document, String countryHint) {
-        Objects.requireNonNull(document, "'document' cannot be null.");
-        return detectLanguageBatch(Collections.singletonList(document), countryHint, null)
-            .map(detectLanguageResult -> {
-                if (detectLanguageResult.isError()) {
-                    throw logger.logExceptionAsError(
-                        Transforms.toTextAnalyticsException(detectLanguageResult.getError()));
-                }
-                return detectLanguageResult.getPrimaryLanguage();
-            }).last();
+        try {
+            Objects.requireNonNull(document, "'document' cannot be null.");
+            return detectLanguageBatch(Collections.singletonList(document), countryHint, null)
+                .map(detectLanguageResult -> {
+                    if (detectLanguageResult.isError()) {
+                        // TODO: throw HttpResponseException instead
```

Comment:
Are you trying to fit this todo in your pr?

Reply:
I think it is doable. https://github.com/Azure/azure-sdk-for-java/issues/10664

Reply:
It will no longer an issue now. It will throw TextAnalyticsException but not HttpResponseException. Because HttpResponseException can only be raised when network call failed. In this case,  it is not.

Reply:
@iscai-msft Wouldn't that apply to Python as well? It not being a service failure i.e not an HttpResponseException?